src/views/aspect/AspectsView/AspectsView.tsx

```diff
@@ -73,21 +73,25 @@ const AspectsView: FC<AspectsViewProps> = ({
             </CardLayoutContainer>
           ) : (
             <CardFilter data={aspects} tagsValueGetter={aspectTagsValueGetter} valueGetter={aspectValueGetter}>
-              {filteredAspects => (
-                <CardLayoutContainer>
-                  {filteredAspects.map((x, i) => (
-                    <CardLayoutItem key={i}>
-                      <AspectCard
-                        aspect={x}
-                        hubNameId={hubNameId}
-                        challengeNameId={challengeNameId}
-                        opportunityNameId={opportunityNameId}
-                        onDelete={x.canDelete ? onDelete : undefined}
-                      />
-                    </CardLayoutItem>
-                  ))}
-                </CardLayoutContainer>
-              )}
+              {filteredAspects =>
+                !filteredAspects.length ? (
+                  t('pages.contribute.no-aspects')
+                ) : (
+                  <CardLayoutContainer>
+                    {filteredAspects.map((x, i) => (
+                      <CardLayoutItem key={i}>
```

Comment:
Index keys should be avoided - not relying on element's position is what keys were introduced for in the first place.

Reply:
I mean the purpose of `key` is to provide React with a more stable identifier than the element's index.